frontend/common/services/useSegment.ts

```diff
@@ -0,0 +1,71 @@
+  endpoints: (builder) => ({
+    getSegments: builder.query<Res['segments'], Req['getSegments']>({
+      query: ({projectId,...rest}) => ({
+        url: `projects/${projectId}/segments/?${Utils.toParam(rest)}`,
+      }),
+      providesTags:(q,e,arg)=>[{ type: 'Segment', id: 'LIST'+arg.projectId },],
+    }),
+    deleteSegment: builder.mutation<Res['segment'], Req['deleteSegment']>({
+      query: (query: Req['deleteSegment']) => ({
+        url: `projects/${query.projectId}/segments/${query.id}`,
+        method: 'DELETE',
+        body: query,
+      }),
+      invalidatesTags: (q,e,arg)=>[{ type: 'Segment', id: 'LIST'+arg.projectId },],
+    }),
+    updateSegment: builder.mutation<Res['segment'], Req['updateSegment']>({
+      query: (query: Req['updateSegment']) => ({
+        url: `projects/${query.projectId}/segments/${query.id}`,
+        method: 'PUT',
+        body: query.segment,
+      }),
+      invalidatesTags:(q,e,arg)=>[{ type: 'Segment', id: 'LIST'+arg.projectId },],
+    }),
+    createSegment: builder.mutation<Res['segment'], Req['createSegment']>({
+      query: (query: Req['createSegment']) => ({
+        url: `projects/${query.projectId}/segments`,
+        method: 'POST',
+        body: query,
+      }),
+      invalidatesTags: (q,e,arg)=>[{ type: 'Segment', id: 'LIST'+arg.projectId },],
+    }),
+    // END OF ENDPOINTS
+  }),
+ })
+
+export async function getSegments(store: any, data: Req['getSegments'], options?: Parameters<typeof segmentService.endpoints.getSegments.initiate>[1]) {
+  store.dispatch(segmentService.endpoints.getSegments.initiate(data,options))
+  return Promise.all(store.dispatch(segmentService.util.getRunningQueriesThunk()))
+}
+  export async function deleteSegment(store: any, data: Req['deleteSegment'], options?: Parameters<typeof segmentService.endpoints.deleteSegment.initiate>[1]) {
+  store.dispatch(segmentService.endpoints.deleteSegment.initiate(data,options))
+    return Promise.all(store.dispatch(segmentService.util.getRunningQueriesThunk()))
+}
+  export async function updateSegment(store: any, data: Req['updateSegment'], options?: Parameters<typeof segmentService.endpoints.updateSegment.initiate>[1]) {
+  store.dispatch(segmentService.endpoints.updateSegment.initiate(data,options))
+    return Promise.all(store.dispatch(segmentService.util.getRunningQueriesThunk()))
+}
+  export async function createSegment(store: any, data: Req['createSegment'], options?: Parameters<typeof segmentService.endpoints.createSegment.initiate>[1]) {
+  store.dispatch(segmentService.endpoints.createSegment.initiate(data,options))
+    return Promise.all(store.dispatch(segmentService.util.getRunningQueriesThunk()))
+}
+  // END OF FUNCTION_EXPORTS
+
+export const {
+  useGetSegmentsQuery,
+  useDeleteSegmentMutation,
+  useUpdateSegmentMutation,
+  useCreateSegmentMutation,
+  // END OF EXPORTS
+} = segmentService
+
+// const { data, isLoading } = useGetSegmentsQuery({ id: 2 }, {}) get hook
```

Comment:
How come we've got a bunch of commented out code here?

Reply:
Usage example

Reply:
Ah, ok - the other comments with usage examples seem to have a 'usage example' prefix. e.g. https://github.com/Flagsmith/flagsmith/pull/1708/files#diff-1f3cc3a02c13bfbc2f59ddf91d03fd4a29cfc247830f1c0082bbd7ee2176215dR71

Reply:
I'll adjust, the CLI generates this so I'll adjust that too - I do prefer that format yeah